accepted/future-releases/constructor-tearoffs/faq.md

```diff
@@ -0,0 +1,256 @@
+
+**Q:** If I can instantiate a generic function, and `List.filled` is a generic function tear-off, why is `List.filled<int>` invalid?
+
+**A:** Because we say so. We could make it mean `(List.filled)<int>`, but we prefer to reserve the syntax for if/when we add proper generic constructors. And you *can* write the parentheses, just please don't. Prefer `List<int>.filled` if that's what you mean&mdash;and if it's not what you mean, you probably can't yet do what you mean.
+
+**Q:** So where can I write an explicit type argument instantiation.
+
+**A:** In short, after an expression *e*, where the type arguments are not then followed by an argument list (then it's just a generic invocation, not an instantiation), and where *e*:
+
+* Denotes a generic class, mixin or type alias (so it's a plain or qualified identifier resolving to a class, mixin or type alias declaration). If followed by `.name`, that name must denote a constructor, otherwise the result is an instantiated `Type` object.
+* Denotes a generic local, static or top-level function declaration (so it's a plain or qualified identifier resolving to such a function declaration). In that case the result is an instantiated tear-off, like an implicit tear-off now, but without relying on inference to find the types.
+* Denotes a generic instance method (so *e* has the form <code>*e*<sub>2</sub>.name</code>) where *e*<sub>2</sub> has an interface type with generic a `name` method. In that case the result is an instantiated instance method tear-off, like an implicit tear-off now, but without relying on inference to find the types.
+* Has a static type which is a callable object type (the static type of *e* is an interface type with a generic `call` method). In that case, <code>*e*\<typeArgs></code> is equivalent to <code>*e*.call\<typeArgs></code>.
+* Or none of the above and *e* has a static type which is a generic function type. In that case we do an instantiation of the generic function value to create a non-generic function.
+
+In all other cases, it's an error.
+
+**Q:** Can I do an explicit instantiation on a `dynamic` value?
+
+**A:** No. We do allow you to *invoke* a `dynamic` value *v* as <code>*v*\<typeArgs>(args)</code>, but we do *not* allow <code>*v*\<typeArgs></code> as a dynamic instantiation. It's not one of the cases above because they all require or imply having a non-`dynamic` static type. (Same for expressions with static type `Function` or `Never`). In short, we need to know the static type of the type arguments before we will try to instantiate them.
+
+**Q:** If `<typeArgs>` is a selector, can't I write `foo<int><int>`?
+
+**A:** Nice try, but no. Since the token after the first `<int>` is `<`, and not one of the tokens listed above, the first `<` is parsed as a less-than operator, and then parsing will fail glamorously when it reaches the `>`.
```

Comment:
Possible follow-up sentence you might want to consider:

> You could of course try doing `(foo<int>)<int>` but there's no way that would work in practice, because the type of `foo<int>` would be a non-generic function.

Reply:
Done.